### PR TITLE
Minor Syntax improvements on wallet migration

### PR DESCRIPTION
**Pull request by hcho112**
This PR contains minor syntax improvements that were commented on:
[…logout step](https://github.com/near/near-wallet/pull/2996)

### CODE REVIEW

File: packages/frontend/src/components/wallet-migration/WalletMigration.jsx [hunk 1]
```diff
@@ -103,18 +104,30 @@ const WalletMigration = ({ open, onClose }) => {
 
     const onLogout = () => {
         setIsLoggingOut(true);
-        return Promise.all(availableAccounts.map(async (accountId) => {
+       
+        return Promise.allSettled(availableAccounts.map(async (accountId) => {
```
Comment (andy-haynes) [inline packages/frontend/src/components/wallet-migration/WalletMigration.jsx:108]: Seeing this laid out I think this is probably clearer to make `onLogout` an `async` method and implement the key deletes in a `for` loop, i.e.

```js
const failedAccounts = [];
for (const accountId of availableAccounts) {
  const publicKey = await wallet.getPublicKey(accountId);
  try {
    const account = await wallet.getAccount(accountId);
    await account.deleteKey(publicKey);
  } catch {
    failedAccounts.push(accountId);
  }
  // do we still want to delete the signer if there was an error?
  // if they don't delete it here and now then when would they?
  await wallet.removeWalletAccount(accountId);
}
...
if (failedAccounts.length) {
  dispatch(showCustomAlert({
    success: false,
    messageCodeHeader: 'error',
    errorMessage: `fail to delete keys for account(s) ${failedAccounts.join(', ')}`,
  }));
}
```

From here I'm not sure what we actually want to _do_ with these, if it really is RPC timeout then you'd probably want to retry before deleting your keystore. I don't think it needs to be addressed for the first cut, we can see if it comes up in the wild.

**Review by andy-haynes — CHANGES_REQUESTED** [review]
Looks good, I think we can simplify it a bit and only show the accounts that failed then we're good to go.

File: packages/frontend/src/components/wallet-migration/WalletMigration.jsx (hunk 1 above)
Comment (hcho112) [reply packages/frontend/src/components/wallet-migration/WalletMigration.jsx]: Totally make sense! much better. 

In terms of deleting the key from the localStorage, I have shift the `await wallet.removeWalletAccount(accountId);` to only delete if it doesn't fail.

**Review by andy-haynes — APPROVED** [review]
Great work!

File: packages/frontend/src/components/wallet-migration/WalletMigration.jsx (hunk 1 above)
Comment (andy-haynes) [reply packages/frontend/src/components/wallet-migration/WalletMigration.jsx]: Good call, keeping their account active will let them know there's still some cleanup to do. Probably not something we'll see come up often but if it does at least it won't interfere with cleanup of other accounts.

**Pull request merged**